airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/poetry/publish/context.py

```diff
@@ -0,0 +1,95 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import os
+from datetime import datetime
+from typing import Optional
+
+from pipelines.airbyte_ci.connectors.context import PipelineContext
+from pipelines.airbyte_ci.connectors.publish.context import PublishConnectorContext
+
+
+class PyPIPublishContext(PipelineContext):
+    def __init__(
+        self,
+        pypi_token: str,
+        package_path: str,
+        report_output_prefix: str,
+        is_local: bool,
+        git_branch: str,
+        git_revision: str,
+        ci_report_bucket: Optional[str] = None,
+        registry: Optional[str] = None,
+        gha_workflow_run_url: Optional[str] = None,
+        dagger_logs_url: Optional[str] = None,
+        pipeline_start_timestamp: Optional[int] = None,
+        ci_context: Optional[str] = None,
+        ci_gcs_credentials: Optional[str] = None,
+        package_name: Optional[str] = None,
+        version: Optional[str] = None,
+    ) -> None:
+        self.pypi_token = pypi_token
+        self.registry = registry or "https://pypi.org/simple"
+        self.package_path = package_path
+        self.package_name = package_name
+        self.version = version
+
+        pipeline_name = f"Publish PyPI {package_path}"
+
+        super().__init__(
+            pipeline_name=pipeline_name,
+            report_output_prefix=report_output_prefix,
+            ci_report_bucket=ci_report_bucket,
+            is_local=is_local,
+            git_branch=git_branch,
+            git_revision=git_revision,
+            gha_workflow_run_url=gha_workflow_run_url,
+            dagger_logs_url=dagger_logs_url,
+            pipeline_start_timestamp=pipeline_start_timestamp,
+            ci_context=ci_context,
+            ci_gcs_credentials=ci_gcs_credentials,
+        )
+
+    @staticmethod
+    async def from_publish_connector_context(connector_context: PublishConnectorContext) -> Optional["PyPIPublishContext"]:
```

Comment:
```suggestion
    from future import __annotations__
    ...
    async def from_publish_connector_context(connector_context: PublishConnectorContext) -> Optional[PyPIPublishContext]:
```

Reply:
Not sure I understand the change - could you explain?

Reply:
The `from future import __annotations__` allows you to add type hints for a local class that has not yet been evaluated. Quoting the class name works otherwise, but I find it cleaner to not quote it.